xonsh/jobs.py

```diff
@@ -153,12 +159,16 @@ def _give_terminal_to(pgid):
                 LIBC.sigprocmask(ctypes.c_int(signal.SIG_SETMASK),
                                  ctypes.byref(omask), None)
     else:
-        def _give_terminal_to(pgid):
-            st = _shell_tty()
-            if st is not None and os.isatty(st):
-                oldmask = signal.pthread_sigmask(signal.SIG_BLOCK,
-                                                 _block_when_giving)
-                os.tcsetpgrp(st, pgid)
+        def give_terminal_to(pgid):
+            oldmask = signal.pthread_sigmask(signal.SIG_BLOCK, _block_when_giving)
+            try:
+                os.tcsetpgrp(sys.stderr.fileno(), pgid)
```

Comment:
I think it is best to stick with `_shell_tty()` here. Is there a reason that didn't work?